plugin/main/src/kotlinx/benchmark/gradle/SuiteSourceGenerator.kt

```diff
@@ -139,6 +139,8 @@ class SuiteSourceGenerator(val title: String, val module: ModuleDescriptor, val
             .filterIsInstance<PropertyDescriptor>()
             .filter { it.annotations.any { it.fqName.toString() == paramAnnotationFQN } }
 
+        validateParameterProperties(parameterProperties)
```

Comment:
Currently, neither corresponding annotations have documentation describing restrictions applied to annotated functions/properties/classes, nor does a doc on writing benchmarks mention it.

I believe that we should not only provide a detailed error message to help users in spotting a problem (which you already did!), but also describe how not to write benchmarks, so that they don't do that in the first place.

Reply:
I've mentioned the restrictions in the doc on writing benchmarks. Since the annotations have no documentation right now, I believe they should be added in a separate PR.